web/src/components/core/ExpandableSelector.jsx

```diff
@@ -0,0 +1,240 @@
+ * @property {string} name - The column header text.
+ * @property {(object) => React.ReactNode} value - A function receiving
+ *   the item to work with and returning the column value.
+ * @property {string} [classNames] - space-separated list of additional CSS class names.
+ */
+
+/**
+ * Internal component for building the table header
+ *
+ * @param {object} props
+ * @param {ExpandableSelectorColumn[]} props.columns
+ */
+const TableHeader = ({ columns }) => (
+  <Thead>
+    <Tr>
+      <Th />
+      <Th />
+      { columns?.map((c, i) => <Th key={i} className={c.classNames}>{c.name}</Th>) }
+    </Tr>
+  </Thead>
+);
+
+/**
+ * Helper function for ensuring a good value for ExpandableSelector#itemsSelected prop
+ *
+ * It logs information to console.error if given value does not match
+ * expectations.
+ *
+ * @param {*} selection - The value to check.
+ * @param {boolean} allowMultiple - Whether the returned collection can have
+ *   more than one item
+ * @return {Array} Empty array if given value is not valid. The first element if
+ *   it is a collection with more than one but selector does not allow multiple.
+ *   The original value otherwise.
+ */
+const sanitizeSelection = (selection, allowMultiple) => {
+  if (!Array.isArray(selection)) {
+    console.error("`itemSelected` prop must be an array. Ignoring given value", selection);
+    return [];
+  }
+
+  if (!allowMultiple && selection.length > 1) {
+    console.error(
+      "`itemsSelected` prop can only have more than one item when selector `isMultiple`. " +
+        "Using only the first element"
+    );
+
+    return [selection[0]];
+  }
+
+  return selection;
+};
+
+/**
+ * Build a expandable table with selectable items
+ * @component
+ *
+ * @note It only accepts one nesting level.
+ *
+ * @param {object} props
+ * @param {ExpandableSelectorColumn[]} props.columns - Collection of objects defining columns.
+ * @param {boolean} [props.isMultiple=false] - Whether multiple selection is allowed.
+ * @param {object[]} props.items - Collection of items to be rendered.
+ * @param {string} [props.itemIdKey="id"] - The key for retrieving the item id.
+ * @param {(item: object) => Array<object>} [props.itemChildren=() => []] - Lookup method to retrieve children from given item.
+ * @param {(item: object) => boolean} [props.itemSelectable=() => true] - Whether an item will be selectable or not.
+ * @param {(item: object) => (string|undefined)} [props.itemClassNames=() => ""] - Callback that allows adding additional CSS class names to item row.
+ * @param {object[]} [props.itemsSelected=[]] - Collection of selected items.
+ * @param {string[]} [props.initialExpandedKeys=[]] - Ids of initially expanded items.
+ * @param {(selection: Array<object>) => void} [props.onSelectionChange=noop] - Callback to be triggered when selection changes.
+ * @param {object} [props.tableProps] - Props for {@link https://www.patternfly.org/components/table/#table PF/Table}.
+ */
```

Comment:
Add a note why this cannot be build with core/TreeTable.

Try to unify them as much as possible (after current milestone)